modules/ting_search_carousel/plugins/content_types/carousel.inc

```diff
@@ -31,7 +36,7 @@ function ting_search_carousel_carousel_content_type_render($subtype, $conf, $pan
       $carousels[] = array(
         '#type' => 'ding_carousel',
         '#title' => $search['title'],
-        '#path' => 'ting_search_carousel/results/ajax/' . $index,
+        '#path' => 'ting_search_carousel/results/ajax/' . urlencode($search['query']),
```

Comment:
Will this url encoding of the search query not give problems with max length in urls as the editors makes crazy long search queries (knowns from e.g eReolen that it already breaks).

Reply:
Would it be possible to create an hash in cache or something that could be used to id the search query for a given carousel?

Reply:
I see that the query input field is set to an max of 2048, but that should be minus the domain name and "'ting_search_carousel/results/ajax/'" to ensure that this don't happen.

Reply:
I am for the second option "validation on the admin form that errors out if the ajax path (after URL encoding) exceeds 2048". This ensures that we preserve HTTP semantics (idempotent requests should use GET) which is also good for caching, which is important in these scenarios.

Editors making crazy long textual queries are in my opinion expressing a need which is currently not handled very well by the system. If we want to address that need I think we should consider what other options there might be.